target/linux/realtek/files-5.10/arch/mips/rtl838x/prom.c

```diff
@@ -193,15 +215,19 @@ void __init prom_init(void)
 
 	prom_init_cmdline();
 
-#ifdef  CONFIG_MIPS_CPS
-	if (!register_cps_smp_ops()) {
+	if (!register_cps_smp_ops())
 		return;
-	}
-#endif
+
 #ifdef CONFIG_MIPS_MT_SMP
-	if (!register_vsmp_smp_ops()) {
+	if (cpu_has_mipsmt) {
+		struct plat_smp_ops rtl_smp_ops;
```

Comment:
@plappermaul Does it work without the static keyword?
On lantiq target kernel then can't boot.

Reply:
I don't get the point. You recommended to reduce the scope of the variable. Is this wrong?

Reply:
After testing on the hardware, I think my recommendation was wrong. In fact, I don't know why it doesn't work after limiting the scope of the variable.